src/map/unit.cpp

```diff
@@ -2451,7 +2416,8 @@ int unit_attack(struct block_list *src,int target_id,int continuous)
 		return 0;
 	}
 
-	if( !unit_can_attack(src, target_id) ) {
+	sc = status_get_sc(src);
+	if( (sc && sc->cant.attack) || !unit_can_attack(src, target_id) ) {
```

Comment:
Why is sc->cant.attack not checked inside unit_can_attack directly?
Both calls to the function do it either right before or right after.